src/ape/api/transactions.py

```diff
@@ -183,26 +184,45 @@ def __repr__(self) -> str:
         return f"<{cls_name} {params}>"
 
     def __str__(self) -> str:
-        # NOTE: Using JSON mode for style.
-        data = self.model_dump(mode="json")
-        if len(data["data"]) > 9:
-            # only want to specify encoding if data["data"] is a string
-            if isinstance(data["data"], str):
-                data["data"] = (
+        return self.to_string()
+
+    def to_string(self, calldata_repr: Optional["CalldataRepr"] = None) -> str:
+        """
+        Get the stringified representation of the transaction.
+
+        Args:
+            calldata_repr (:class:`~ape.types.abi.CalldataRepr` | None): Pass "full"
+              to see the full calldata. Defaults to the value from the config.
+
+        Returns:
+            str
+        """
+        data = self.model_dump(mode="json")  # JSON mode used for style purposes.
+
+        if calldata_repr is None:
+            # If was not specified, use the default value from the config.
+            calldata_repr = self.local_project.config.calldata_repr
```

Comment:
What do you think about namespacing it under `display: ` key? In case we add more core configuration for display purposes later?

Reply:
done